WordPress/src/main/java/org/wordpress/android/ui/posts/PostListEventListener.kt

```diff
@@ -128,13 +128,15 @@ class PostListEventListener(
                 )
             }
             is CauseOfOnPostChanged.RemoteAutoSavePost -> {
+                val post = postStore.getPostByLocalPostId((event.causeOfChange as RemoteAutoSavePost).localPostId)
                 if (isRemotePreviewingFromPostsList.invoke()) {
-                    val post = postStore.getPostByLocalPostId((event.causeOfChange as RemoteAutoSavePost).localPostId)
                     if (event.isError) {
                         AppLog.d(T.POSTS, "REMOTE_AUTO_SAVE_POST failed: " +
                                 event.error.type + " - " + event.error.message)
                     }
                     handleRemoteAutoSave(post, event.isError)
+                } else {
+                    uploadStatusChanged(post.id)
```

Comment:
We need to inform the UI that the upload has finished so it can hide the progress bar.

Reply:
I've experienced rare instances where the progress bar just gets stuck. I, unfortunately, could not find consistent reproducible steps. 😞